tools/src/main/scala/org/apache/spark/sql/rapids/tool/qualification/QualificationAppInfo.scala

```diff
@@ -303,18 +355,32 @@ class QualificationAppInfo(
       // wall clock time
       val executorCpuTimePercent = calculateCpuTimePercent(perSqlStageSummary)
 
-      // TODO - do we want to use this and rely on stages, but some SQL don't have stages
-      // so this is less than SQL DF real
-      val sqlDFWallClockDuration =
-        perSqlStageSummary.map(s => s.hackEstimateWallclockSupported).sum
-      val allStagesSummary = perSqlStageSummary.map(_.stageSum)
-      val sqlDataframeTaskDuration = allStagesSummary.flatMap(_.map(s => s.stageTaskTime)).sum
+      // Using the spark SQL reported duration, this could be a bit off from the
+      // task times because it relies on the stage times and we might not have
+      // a stage for every exec
+      val allSQLDurations = sqlIdToInfo.map { case (_, info) =>
+        info.duration.getOrElse(0L)
+      }
+      val sparkSQLDFWallClockDuration = allSQLDurations.sum
+      val longestSQLDuration = if (allSQLDurations.size > 0) {
+        allSQLDurations.max
+      } else {
+        0L
+      }
+
+      val sqlDFSupportedWallClockDurationBasedStages = perSqlStageSummary.map { s =>
```

Comment:
This variable is not used in any further calculation. Should it be removed OR should we take max of this and `sparkSQLDFWallClockDuration` while passing  in `QualificationAppInfo.calculateEstimatedInfoSummary` method ?